crates/oxc_semantic/src/checker/javascript.rs

```diff
@@ -293,10 +293,12 @@ pub fn check_number_literal(lit: &NumericLiteral, ctx: &SemanticBuilder<'_>) {
 
     if ctx.strict_mode() {
         match lit.base {
-            NumberBase::Octal if leading_zero(lit.raw) => {
+            NumberBase::Octal if leading_zero(lit.raw.as_ref().unwrap().as_str()) => {
                 ctx.error(legacy_octal(lit.span));
             }
-            NumberBase::Decimal | NumberBase::Float if leading_zero(lit.raw) => {
+            NumberBase::Decimal | NumberBase::Float
+                if leading_zero(lit.raw.as_ref().unwrap().as_str()) =>
+            {
```

Comment:
Semantic checker can be run on an AST which has been through transformer or minifier, or otherwise programatically generated. So it's possible `raw` is `None`, and shouldn't unwrap.

Here I think solution is just to skip the checks if `raw` is `None`. e.g.:

```rs
fn leading_zero(s: &Option<Atom>) -> bool {
    if let Some(s) = s.as_ref() {
        let mut chars = s.bytes();
        if let Some(first) = chars.next() {
            if let Some(second) = chars.next() {
                return first == b'0' && second.is_ascii_digit();
            }
        }
    }
    false
}

if ctx.strict_mode() {
    match lit.base {
        NumberBase::Octal if leading_zero(&lit.raw) => {
            ctx.error(legacy_octal(lit.span));
        }
        NumberBase::Decimal | NumberBase::Float if leading_zero(&lit.raw) => {
            ctx.error(leading_zero_decimal(lit.span));
        }
        _ => {}
    }
}
```